mscore/plugin/pluginUpdater.cpp

```diff
@@ -0,0 +1,994 @@
+                                                      desc.qml_paths.push_back(e.readElementText());
+                                                else
+                                                      e.unknown();
+                                                }
+                                          }
+                                    else if (t == "GitHubReleaseID")
+                                          desc.release_id = e.readInt();
+                                    else if (t == "GitHubLatestSha")
+                                          desc.latest_commit = e.readElementText();
+                                    else if (t == "LastModified")
+                                          desc.last_modified = QDateTime::fromString(e.readElementText());
+                                    else
+                                          e.unknown();
+                                    }
+                              if (!page_url.isEmpty())
+                                    pluginDescriptionMap.insert(page_url, desc);
+                              else
+                                    qDebug("Missing plugin page url.");
+                              }
+                        else
+                              e.unknown();
+                        }
+                  }
+            else
+                  e.unknown();
+            }
+      return true;
+      }
+
+PluginWorker::PluginWorker(ResourceManager* r) : r(r)
+      {
+      QObject::connect(this, SIGNAL(pluginStatusChanged(int, PluginStatus)), r, SLOT(refreshPluginButton(int, PluginStatus)));
+      }
+
+PluginWorker::PluginWorker(PluginPackageDescription& desc, ResourceManager* r) : desc(desc), r(r)
+      {
+      QObject::connect(this, SIGNAL(pluginStatusChanged(int, PluginStatus)), r, SLOT(refreshPluginButton(int, PluginStatus)));
+      }
+
+void PluginWorker::checkUpdate(QPushButton* install)
+      {
+      const QString& page_url = install->property("page_url").toString();
+      PluginPackageDescription* desc_tmp = new PluginPackageDescription(desc);
+      PluginPackageDescription desc_backup = desc;
+      install->setEnabled(false);
+      install->setText("Checking for update…");
+      bool should_update = analyzePluginPage("https://musescore.org" + page_url);
+      // very hack here. should be improved.
+      *desc_tmp = desc;
+      desc = desc_backup;
+      if (should_update) {
+            desc.update = desc_tmp;
+            r->commitPlugin(page_url, desc);
+            emit pluginStatusChanged(install->property("row").toInt(), PluginStatus::UPDATE_AVAILABLE);
+            }
+      else {
+            delete desc_tmp;
+            emit pluginStatusChanged(install->property("row").toInt(), PluginStatus::UPDATED);
+            }
+      }
+
+void PluginWorker::timeconsume()
+      {
+      qDebug("timeconsume start");
+      int member = 0;
+      for (int i = 0; i < 1000000; i++)
+            member = member * 2 + 1;
```

Comment:
wouldn't that overflow pretty quick?
And wouldn't any decent optimizer remove that whole loop?

Reply:
This is a function I previously made time-consuming and used to test multi-threading, and should be removed. Sorry for the confusion.